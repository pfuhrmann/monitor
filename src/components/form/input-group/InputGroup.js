import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './InputGroup.scss';

const InputGroup = ({ children, className, ...otherProps }) => (
  <div className={cx('Form-input-group', className)} {...otherProps}>
    { children }
  </div>
);

InputGroup.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

InputGroup.defaultProps = {
  children: '',
  className: '',
};

export default InputGroup;
