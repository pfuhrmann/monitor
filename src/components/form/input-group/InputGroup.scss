@import "../../../variables.scss";

.Form-input-group {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding-right: 2em;
}
