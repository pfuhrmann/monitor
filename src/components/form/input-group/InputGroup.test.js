import React from 'react';
import ReactDOM from 'react-dom';
import InputGroup from './InputGroup';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<InputGroup><input /></InputGroup>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders input group correctly', () => {
  const component = renderer.create(
    <InputGroup><input /></InputGroup>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
