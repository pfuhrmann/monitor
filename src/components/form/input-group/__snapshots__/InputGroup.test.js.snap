// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input group correctly 1`] = `
<div
  className="Form-input-group"
>
  <input />
</div>
`;
